### PR TITLE
Documentation change to macros.rs for `includes!`

### PR DESCRIPTION
**Pull request by alexispurslane**
I'm not sure if this documentation is clear or extensive enough, but this is just to get started on the problem, fixes issue #36387.

### CODE REVIEW

File: src/libstd/macros.rs [hunk 1]
```diff
@@ -444,9 +444,16 @@ pub mod builtin {
     #[macro_export]
     macro_rules! cfg { ($($cfg:tt)*) => ({ /* compiler built-in */ }) }
 
-    /// Parse the current given file as an expression.
-    ///
-    /// This is generally a bad idea, because it's going to behave unhygienically.
+    /// Parse the file provided in the argument as an expression, item, or
+    /// statement according to the context. This file is located relative to the
```
Comment (jseyfried) [inline src/libstd/macros.rs]: I believe the file is can only be parsed as an expression or an item -- c.f. [this `impl`](https://github.com/rust-lang/rust/blob/79fa9eb643cfbb807813afed6f825f6654ee7662/src/libsyntax/ext/source_util.rs#L108).

File: src/libstd/macros.rs [hunk 2]
```diff
@@ -444,9 +444,16 @@ pub mod builtin {
...
+    /// current file (similarly to how modules are found).
+    ///
+    /// Using this macro is often a bad idea, because the file is
+    /// going to be parsed and placed in the surrounding code
+    /// unhygenically. This could result in variables or functions
+    /// being different from what the file expected if there are
+    /// variables or functions that have the same name in the current
+    /// file.
```
Comment (jseyfried) [inline src/libstd/macros.rs]: If the file as parsed as an item, I believe hygiene is never an issue (that is, using `include!`ing items is equivalent to copying and pasting source code).

**Comment by aturon** [comment]
Apologies for the very delayed review, and thanks for the PR!

@bors: r+ rollup

**Pull request merged**